pkg/route/api/validation/validation.go

```diff
@@ -38,7 +40,7 @@ func ValidateRoute(route *routeapi.Route) field.ErrorList {
 		}
 	}
 
-	if err := validateWildcardPolicy(route.Spec.Host, route.Spec.WildcardPolicy, specPath.Child("wildcardPolicy")); err != nil {
+	if err := validateWildcardPolicy(route.Spec.Host, route.Spec.WildcardPolicy, route.Annotations[HostGeneratedAnnotationKey] == "true", specPath.Child("wildcardPolicy")); err != nil {
```

Comment:
I would expect the strategy to not auto-generate a host when wildcardPolicy=Subdomain, and validation to require a non-empty host when wildcardPolicy=Subdomain. Then you don't have to make the annotation govern validation, and they get a message about host being required when wildcardPolicy=Subdomain, which is accurate.